lib_xcore_math/src/bfp/bfp_s16.c

```diff
@@ -339,12 +339,10 @@ float bfp_s16_mean(
     headroom_t hr = HR_S32(sum) + 32;
     int64_t sum64 = ((int64_t)sum) << hr;
     int64_t mean64 = sum64 / ((int)b->length);
-    right_shift_t shr = MAX(0, 48 - HR_S64(mean64));
+    right_shift_t shr = MAX(0, 32 - HR_S64(mean64));
```

Comment:
I suppose this was the bug you found, well done!

Reply:
This was not a bug but, it's resolving a TODO from below.